specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1287 @@
+        // to add arguments to the button.
+        Button(ArgumentSerializer serializer);
+        Button(String content, ArgumentSerializer serializer);
+
+        // Sets the IconUri for the button.
+        Button SetIconUri(Windows.Foundation.Uri iconUri);
+
+        // The tooltip for a button, if the button has an empty content string.
+        Button SetToolTip(String toolTip);
+
+        // Sets the Button as context menu action.
+        Button UsesContextMenuPlacement(); // AllowContextMenuPlacement
+
+        // Sets the ButtonStyle to Success or Critical
+        Button SetButtonStyle(ButtonStyle buttonStyle);
+
+        // Sets the ID of an existing TextBox in order to have this button display to the right of the input.
+        Button SetInputId(String inputId);
+
+        // Retrieves the XML content of the button.
+        String GetXml();
+    };
+
+    runtimeclass TextBox
+    {
+        TextBox(String id);
+
+        // The placeholder displayed for text input.
+        TextBox SetPlaceHolderContent(String placeHolderContent);
+
+        // Retrieves the XML content of the input.
+        String GetXml();
+    };
+
+    runtimeclass SelectionMenu
+    {
+        SelectionMenu(String id);
+
+        // Add a selection item to the SelectionMenu if InputType is SelectionMenu. The id parameter is used to retrieve the user input when the app is activated
+        // and content is the displayed text for the item.
+        SelectionMenu AddSelection(String id, String content);
+
+        // Sets the default selection to be displayed by the SelectionMenu
+        SelectionMenu SetDefaultSelection(String id);
+
+        // Retrieves the XML content of the input.
+        String GetXml();
+    };
+
+    runtimeclass ProgressBar
+    {
+        // ProgressBar binds to AppNotificationProgressData so the AppNotification will
+        // receive every update to the status and value. In the WinAppSDK, these binding
+        // values are static, so developers won't need to define these binding values
+        // themselves.
+        ProgressBar();
+
+        // Adds the title binding value.
+        ProgressBar UseTitle();
+
+        // Adds the valueStringOverride value.
+        ProgressBar UseValueStringOverride();
+
+        // Retrieves the XML content of the ProgressBar
+        String GetXml();
+    };
+
+    enum MSWinSoundEvent
+    {
+        Default,
+        IM,
+        Mail,
+        Reminder,
+        SMS,
+        Alarm,
+        Alarm2,
+        Alarm3,
+        Alarm4,
+        Alarm5,
+        Alarm6,
+        Alarm7,
+        Alarm8,
+        Alarm9,
+        Alarm10,
+        Call,
+        Call2,
+        Call3,
+        Call4,
+        Call5,
+        Call6,
+        Call7,
+        Call8,
+        Call9,
+        Call10,
+    };
+
+    runtimeclass Audio
+    {
+        Audio();
+        Audio(MSWinSoundEvent msWinSoundEvent);
+        Audio(Windows.Foundation.Uri audioUri);
+
```

Comment:
This won't compile: error MIDL4061: [msg]More than one factory method with same number of arguments in the projected constructors.